library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -0,0 +1,592 @@
+  protected String highestPolled;
+  private static final String user = "user";
+  private static final String password = "password";
+  /**
+   * thread to poll database
+   */
+  private transient Thread dbPoller;
+  protected transient ArrayBlockingQueue<List<T>> emitQueue;
+  protected transient PreparedStatement ps;
+  protected WindowDataManager windowManager;
+  private String emitColumnList;
+
+  public String getWhereCondition()
+  {
+    return whereCondition;
+  }
+
+  public void setWhereCondition(String whereCondition)
+  {
+    this.whereCondition = whereCondition;
+  }
+
+  public String getEmitColumnList()
+  {
+    return emitColumnList;
+  }
+
+  public void setEmitColumnList(String emitColumnList)
+  {
+    this.emitColumnList = emitColumnList;
+  }
+
+  public int getFetchSize()
+  {
+    return fetchSize;
+  }
+
+  public void setFetchSize(int fetchSize)
+  {
+    this.fetchSize = fetchSize;
+  }
+
+  protected abstract void pollRecords(PreparedStatement ps);
+
+  public int getPollInterval()
+  {
+    return pollInterval;
+  }
+
+  public void setPollInterval(int pollInterval)
+  {
+    this.pollInterval = pollInterval;
+  }
+
+  public int getQueueCapacity()
+  {
+    return queueCapacity;
+  }
+
+  public void setQueueCapacity(int queueCapacity)
+  {
+    this.queueCapacity = queueCapacity;
+  }
+
+  public String getKey()
+  {
+    return key;
+  }
+
+  public void setKey(String key)
+  {
+    this.key = key;
+  }
+
+  public String getTableName()
+  {
+    return tableName;
+  }
+
+  public void setTableName(String tableName)
+  {
+    this.tableName = tableName;
+  }
+
+  public KeyValPair<String, String> getRangeQueryPair()
+  {
+    return rangeQueryPair;
+  }
+
+  public void setRangeQueryPair(KeyValPair<String, String> rangeQueryPair)
+  {
+    this.rangeQueryPair = rangeQueryPair;
+  }
+
+  public int getBatchSize()
+  {
+    return batchSize;
+  }
+
+  public void setBatchSize(int batchSize)
+  {
+    this.batchSize = batchSize;
+  }
+
+  public AbstractJdbcPollInputOperator()
+  {
+    this.pollInterval = 10 * 1000;
+    this.fetchSize = 10000;
+    currentWindowRecoveryState = new MutablePair<>();
+    windowManager = new WindowDataManager.FSWindowDataManager();
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    super.setup(context);
+    spinMillis = context.getValue(Context.OperatorContext.SPIN_MILLIS);
+    execute = true;
+    cause = new AtomicReference<Throwable>();
+    emitQueue = new ArrayBlockingQueue<List<T>>(queueCapacity);
+    this.context = context;
+    operatorId = context.getId();
+
+    try {
+      if (ps == null) {
+        //If its a range query pass upper and lower bounds
+        //If its a polling query pass only the lower bound
+        if (getRangeQueryPair().getValue() != null) {
+          ps = store.getConnection().prepareStatement(
+              JdbcMetaDataUtility.buildRangeQuery(getTableName(), getKey(), rangeQueryPair.getKey(),
```

Comment:
Do you need to initialize "ps" here? I see that it is being done in beginWindow() again